Refactor and add filters to members page

chinins opened the pull request:
## Description

This PR adds filters to the members page. It also refactors the code a bit to split and simplify it. I've also added fixed width to the MembersListItem so that watchers item list doesn't shrink when there are no contributors.

Please, note that the hover state of the `reset` button for the filters is not as in the specs. It was agreed with Arren that it will stay the same as all other buttons above.

Update figma specs
https://www.figma.com/file/qQoBbyyDXkuHwoqHIOGjEu/Colony-Simplified?node-id=7705%3A91695

<img width="1028" alt="Screenshot 2022-07-01 at 16 08 23" src="https://user-images.githubusercontent.com/34057551/176913877-da6a345c-5506-4adc-ab04-de4b3565c81f.png">


**New stuff** ✨

- add filters checkboxes and logic

**Changes** 🏗

- extract MemberControls to a separate component
- fix the MembersListItem width

resolves #3381

willm30 reviewed (changes requested):
Nice work! 

I noticed a few things as I was going through:
1. When the team filter is applied, the [invite link should be hidden](https://www.figma.com/file/qQoBbyyDXkuHwoqHIOGjEu/Colony-Simplified?node-id=6736%3A94732)
![invite-menu](https://user-images.githubusercontent.com/64402732/176924616-0df1f81c-36e7-455e-9c5d-e7edfcc5c528.png)

2. There aren't the numbers next to each filter indicating the number of matches
![no-numbd](https://user-images.githubusercontent.com/64402732/176925620-653e7c39-8651-4e3c-bede-6201ba8476e3.png)

3. The filters jump when nothing is selected
![jumpy-filter](https://user-images.githubusercontent.com/64402732/176924669-4bcc1e84-10c4-4914-8050-1ec26c3e819c.gif)

4. I think it would be good to include a simple message here when nothing is selected. Something like "No filters selected." would suffice.
![no-members-msg](https://user-images.githubusercontent.com/64402732/176924609-16bbea3e-7f19-4279-9076-ceb759bac936.png)

ArmandoGraterol reviewed (changes requested):
For me, that I have a large screen to test it, it doesn't jump like the gif that WIll uploaded, however, it does seem like the width doesn't  adapt to the container:
![FireShot Capture 649 - Members - Colony - wonker - localhost](https://user-images.githubusercontent.com/18473896/176938484-04fb1dcf-ea79-4d81-97b5-655f898fe3b6.png)

chinins commented:
@willm30, @ArmandoGraterol, I've fixed the width adjustments. Please, let me know if it's working properly for you.

I've also removed the invite link when it's not a root domain.

As for the count in the filter checkbox labels, it's been decided  with Arren not to have it.

chinins commented:
> @chinins Looks good! Pre-approving. Just flagging the following error that's appearing in the console when loading the members page:

It's not an error, it's a warning. I'm aware of it. It also happens in several places in the dapp so it's not a big deal.

danbr commented on src/modules/dashboard/components/Members/filterMembers.ts:
```diff
@@ -1,21 +1,64 @@
+  /* All the filters together */
+  return data.filter(({ banned, isWhitelisted, profile, id }) => {
+    const textFilter =
+      profile?.username?.toLowerCase().includes(filterValue?.toLowerCase()) ||
```
This block is almost a duplicate of the above, just without `textFilter`. Could this be refactored?

danbr reviewed (approved):
Very good. Works for me. Approving but have a comment (in the code)....

arrenv reviewed (changes requested):
Working well! I noticed the filter doesn't work to show only Verified and Banned.

![download (32)](https://user-images.githubusercontent.com/33682027/177171823-0a822217-4017-41df-9057-f7f5255155ab.png)

![download (33)](https://user-images.githubusercontent.com/33682027/177171840-2558ed6c-07e7-4d53-b2d5-58a06b944f87.png)

![download (34)](https://user-images.githubusercontent.com/33682027/177171847-a06bf574-dab8-4d8d-b039-c1c62ecabc17.png)

ArmandoGraterol commented:
> @willm30, @ArmandoGraterol, I've fixed the width adjustments. Please, let me know if it's working properly for you.
 
It's looking good on my end now! Thank you

arrenv reviewed (approved):
It is working really well, nice one!

willm30 reviewed (changes requested):
Filters looking great! Noticed a couple of things re. functionality: 

1. Not banned contributors are not showing up in "Not banned"
![banned-members](https://user-images.githubusercontent.com/64402732/177589993-1d620e25-64bb-46d8-89b5-2538140363c2.gif)

2. Verifying someone overrides previous verification. To reproduce: 
* Verify somebody and refresh
* Verify somebody else and refresh
* The first person is no longer verified. 

And a tiny thing re container width: 

3. The container shrinks slightly when the filters return no entries
![collapsing-title-members](https://user-images.githubusercontent.com/64402732/177590304-2118d8d5-8383-4f5e-87ec-4463298a64df.gif)

danbr reviewed (commented):
(UN)Fortunately, I dont have the issues that @willm30 has experienced.

For me, the 'Banned' dropdown is at the bottom of page, so clicking dont give any obvious reaction as the dropdown options are off page.
It could be worth considering scrolling the page down when clicking on the 'Banned' dropdown.

(Also not great to have the feedback link on top of the dropdown)

<img width="1125" alt="Screenshot 2022-07-06 at 21 40 17" src="https://user-images.githubusercontent.com/582700/177621348-d2efec69-2b0b-4301-b5a5-30c31782a1e3.png">

ArmandoGraterol reviewed (commented):
For me, the contributor appears when the `Not banned` filter is applied but the watcher is disappearing:
![FireShot Capture 655 - Members - Colony - wonker - localhost](https://user-images.githubusercontent.com/18473896/177667990-c36a8414-3c3e-406f-b61c-85fcae08fcaf.png)

![FireShot Capture 656 - Members - Colony - wonker - localhost](https://user-images.githubusercontent.com/18473896/177667959-652e0137-f36b-4927-8e59-84fc0e4f333f.png)

Other than that, I haven't noticed anything strange

willm30 commented:
I have re-tested this morning. I'm still noticing 2 & 3, so will create a separate issue for 2. 3 (container shrink) occurs when filtering to "Watchers". 

As for banned filter, I believe it's related to the verified address. I.e., for me, when the member is verified, they are not being included at all by the banned filter. Have reproduced again with a different example (**@d is the member to watch**): 

![banned-behaving-off](https://user-images.githubusercontent.com/64402732/177733716-d06a30e5-099e-4f16-8021-c6dac1617bd3.gif)

arrenv commented:
> As for banned filter, I believe it's related to the verified address. I.e., for me, when the member is verified, they are not being included at all by the banned filter. Have reproduced again with a different example (**@d is the member to watch**):

I tested again just now again as well, I am experiencing the same thing with the same combo.

Member type: All members
Verification type: Any
Banned status: Not banned

Member who is verified disappears, when they shouldn't.

chinins commented:
@willm30, @ArmandoGraterol, I've fixed the banned bug.

@danbr, I've added the scroll.

@willm30, as discussed, nobody can reproduce the jumping screen. Please, raise an issue and work on it since you are the only one who experience this problem.

willm30 reviewed (approved):
I have created the following issues to follow up with things I noticed here:

1. [Members List width shrinks when filter watchers](https://github.com/JoinColony/colonyDapp/issues/3593)
(**Note, to reproduce you need to verify the contributor but not the watcher (see issue for set up)**)

2.  [Address book only verifying one member at a time](https://github.com/JoinColony/colonyDapp/issues/3588)

As such, approving as the banned filter is working as expected and the rest works great.

chinins commented:
> I have created the following issues to follow up with things I noticed here:
> 
> 1. [Members List width shrinks when filter watchers](https://github.com/JoinColony/colonyDapp/issues/3593)
>    (**Note, to reproduce you need to verify the contributor but not the watcher (see issue for set up)**)
> 

With this setup, I can reproduce. So it's an issue up for grabs for anybody. Good catch about the difference.

ArmandoGraterol reviewed (approved):
Working great! Excellent job Olga :) Tested a lot of combinations and it's working as expected

danbr reviewed (approved):
The auto scrolling updates work well. It all LGTM.
